lib-extra/src/main/java/com/diffplug/spotless/extra/java/EclipseFormatterStep.java

```diff
@@ -47,14 +40,14 @@ private EclipseFormatterStep() {}
 	private static final String FORMATTER_METHOD = "format";
 
 	/** Creates a formatter step for the given version and settings file. */
-	public static FormatterStep create(File settingsFile, Provisioner provisioner) {
-		return create(defaultVersion(), settingsFile, provisioner);
+	public static FormatterStep create(Iterable<File> settingsFiles, Provisioner provisioner) {
```

Comment:
I would prefer the `@deprecated`, because it transports the goal to the developer: 

1. Use the common user interface if your formatter is configured by properties.
2. FormatterProperties is the central place to load the properties.

When I introduce my tests EclipseFormatterStepTest, I will (if you agree) anyhow refactor the EclipseFormatterStepTest to allign them with the once I have for GrEclipse. Why should I test the load of different property representations (formatter.properties/formatter.xml) in the formatter steps. That has been done already in this commit by the FormatterPropertiesTest.

Looking freshmark, I saw how you deal with property values which can be calculated at run time.
Maybe this can be once incorporated, but I do not know whether people would use that frequently. The code/resource formatting is in itself an encapsulated job. Big lists of configuration items should not mess up you build.gradle.